.github/workflows/on_push_main.yml

```diff
@@ -291,6 +292,17 @@ jobs:
       WHEEL_ARTIFACT_NAME: windows-x64-wheel
     secrets: inherit
 
+  # --------------------------------------------------------------------------
+
+  run-notebook:
```

Comment:
related to above comment, maybe this is better in nightly anyways? I think our on-push-to-main actually does a bit too much as well 🤔